packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -216,7 +220,30 @@ String _validateProjectName(String projectName) {
     return "Invalid project name: '$projectName' - this will conflict with Flutter "
     "package dependencies.";
   }
+  return null;
+}
 
+/// Return `null` if the project directory is legal. Return a validation message
+/// if we should disallow the directory name.
+String _validateProjectDir(String projectName) {
+  var type = FileSystemEntity.typeSync(projectName);
+  if (type != FileSystemEntityType.NOT_FOUND) {
+    switch(type) {
+      case FileSystemEntityType.DIRECTORY:
+        // Do not re-use directory if it is not empty.
+        if (new Directory(projectName).listSync(followLinks: false).isNotEmpty) {
+          return "Invalid project name: '$projectName' - refers to a directory "
+          "that is not empty.";
```

Comment:
Not sure what you mean. The function? It is top-level. The extended string in the return? That's consistent with the other concatenated string in returns so I thought it should be that way.